packages/frontend/src/components/creator/CreatorPoolParamsDialog.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  Button,
+  FormControl,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputRightAddon,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  useToast,
+} from '@chakra-ui/react'
+import { useDeployments } from '@lib/useDeployments'
+import { BeneficiaryPool__factory } from '@yieldgate/contracts/typechain-types'
+import { ethers } from 'ethers'
+import { FC, useRef, useState } from 'react'
+import { useSigner } from 'wagmi'
+
+export interface CreatorPoolParamsDialogProps {
+  isOpen: boolean
+  onClose: () => void
+  poolAddress?: string | false
+  minAmount?: string
+  minDurationDays?: number
+  refetchPoolParams: () => void
+}
+export const CreatorPoolParamsDialog: FC<CreatorPoolParamsDialogProps> = ({
+  isOpen,
+  onClose,
+  poolAddress,
+  minAmount: fetchedMinAmount,
+  minDurationDays: fetchedMinDurationDays,
+  refetchPoolParams,
+}) => {
+  const initialRef = useRef(null)
+  const { data: signer, refetch: refetchSigner } = useSigner()
+  const { contracts, contractsChain } = useDeployments()
+  const [minAmount, setMinAmount] = useState<string>()
+  const [minDurationDays, setMinDurationDays] = useState<number>()
+  const [isLoading, setIsLoading] = useState(false)
+  const toast = useToast()
+
+  const updatePoolParameters = async () => {
+    await refetchSigner()
+    if (!poolAddress || !contracts || !signer) return
+    setIsLoading(true)
+
+    try {
+      const poolContract = BeneficiaryPool__factory.connect(poolAddress, signer)
+      const minDurationSeconds =
+        (minDurationDays ?? Math.floor(fetchedMinDurationDays || 0)) * 24 * 60 * 60
+      const transaction = await poolContract.setParameters(
+        ethers.utils.parseEther(`${minAmount ?? fetchedMinAmount ?? '0.0'}`),
+        minDurationSeconds,
+        {
+          gasLimit: 500000,
```

Comment:
`setParameters` is a rather simple function, so a much lower gas limit can be used here, in this case `80_000`.

In general, it helps running the contract tests with a gas report `REPORT_GAS=true pnpm hardhat test` and then take the max. Of course this only helps for functions that don't make external calls to mocks ^^ `setParams` doesn't interact with other contracts, so we should be able to use it.

![image](https://user-images.githubusercontent.com/2607372/187403472-99a6a1de-300b-4175-ae69-d569cd820efd.png)
This is the current report. We can use the max for `deployPool` and `setParameters`, plus some slack (also for`setMinDuration`&`setMinAmount`, but we don't use these right now).

Once we have integration tests, we may also be able to deduce sensible gas limits for staking, unstaking and claiming.

Reply:
I set more reasonable gas limits in https://github.com/yieldgate/yieldgate/pull/56/commits/9dfddb8164e6d78034d5aaf11f790c8f8cb4c8cb